spec/features/end_deployment_spec.rb

```diff
@@ -0,0 +1,29 @@
+#
+# RPC style seems cleaner than REST here, as setting the endedAt parameter directly
+# seems likely to end in Timezone tears
+# This endpoint would be called by the pact broker client during `record-deployment` if the
+# --end-previous-deployment (on by default) was specified.
+# This allows us to know exactly what is deployed to a particular environment at a given time,
+# (eg. /environments/test/deployments/current)
+# and provides first class support for mobile clients that have multiple versions in prod
+# at once.
```

Comment:
@mefellows thoughts?

Reply:
So if I understand correctly, you first record a deployment is in progress and once the deployment is complete you mark it as done? Or is the `--end-previous-deployment` basically saying, "this deployment replaces the last one"? (I think the latter on reflection, to account for the mobile case to say "this deployment co-exists with the previous one).

Is there merit in capturing that a deployment _attempt_ occurred, and that the deployment itself was successful (or not)?

Having that information (and subsequent screens etc.) might be useful for future CI/CD workflows.

Would you want to

Reply:
`record-deployment` would replace tagging the version with "prod", and would be called after the deployment was done. Yes, the "--end-previous-deployment" would be saying "this one replaces the last one". We can get feedback on a more intuitive name for the flag if that's not clear.

Reply:
> Is there merit in capturing that a deployment attempt occurred, and that the deployment itself was successful (or not)?

It might be useful for *a* product, but I don't think the concern is relevant for *our* product (focussing on application compatibility). Not at this stage, anyway.